src/base/ApplicationDynamoDBTable.ts

```diff
@@ -140,6 +146,52 @@ export class ApplicationDynamoDBTable extends Resource {
       ],
       dependsOn: [targetTracking, dynamoDB],
     });
+
+    // TODO: do we want to assume that all global secondary indexes get a scaling policy?
+    if (globalSecondaryIndexes.length) {
+      globalSecondaryIndexes.forEach((gsIndex) => {
+        // min capacity is defined by the global secondary index
+        // max capacity is inherited from the table auto scaling config
+        // TODO: is this a good forced/default behavior?
+        const minCapacity =
+          capacityType === ApplicationDynamoDBTableCapacityType.Read
+            ? gsIndex.readCapacity
+            : gsIndex.writeCapacity;
+
+        const indexTargetTracking = new AppautoscalingTarget(
```

Comment:
Looks like line 161 here to 193 could be extracted to a method and reused for both table and index autoscaling